src/Callback.php

```diff
@@ -105,88 +91,85 @@ public function init(): void
      */
     public function set($callback, $args = [])
     {
-        if ($this->postTrigger) {
-            if (isset($_POST[$this->postTrigger])) {
-                $this->app->catch_runaway_callbacks = false;
-                $this->triggered = $_POST[$this->postTrigger];
-
-                $t = $this->app->run_called;
-                $this->app->run_called = true;
-                $ret = call_user_func_array($callback, $args);
-                $this->app->run_called = $t;
-
-                return $ret;
-            }
-        } else {
-            if (isset($_GET[$this->urlTrigger])) {
-                $this->app->catch_runaway_callbacks = false;
-                $this->triggered = $_GET[$this->urlTrigger];
-
-                $t = $this->app->run_called;
-                $this->app->run_called = true;
-                $this->owner->stickyGet($this->urlTrigger);
-                $ret = call_user_func_array($callback, $args);
-                //$this->app->stickyForget($this->name);
-                $this->app->run_called = $t;
-
-                return $ret;
-            }
+        if ($this->canTrigger()) {
+            $this->app->catch_runaway_callbacks = false;
+            $t = $this->app->run_called;
+            $this->app->run_called = true;
+            $ret = call_user_func_array($callback, $args);
+            $this->app->run_called = $t;
+
+            return $ret;
         }
     }
 
     /**
      * Terminate this callback
-     * by rendering the owner view.
+     * by rendering the owner view by default.
      */
-    public function terminate()
+    public function terminateJson(View $view = null)
     {
         if ($this->canTerminate()) {
-            $this->app->terminateJson($this->owner);
+            $this->app->terminateJson($view ?: $this->owner);
         }
     }
 
     /**
-     * Prevent callback from terminating during a reload.
+     * Only current callback can terminate.
+     */
+    public function canTerminate(): bool
+    {
+        return $this->urlTrigger === ($_GET['__atk_callback'] ?? null);
+    }
+
+    /**
+     * Allow callback to be triggered or not.
      */
-    protected function canTerminate(): bool
+    public function canTrigger(): bool
```

Comment:
very hard to understand what is the difference between isTrigger, from it's name

I highly prefer no `canTrigger` method - can we move it to `isTrigger`?

Reply:
`canTrigger` is now excl. `isTriggered()`, resolved